.github/workflows/cicd.yml

```diff
@@ -0,0 +1,106 @@
+name: Continuous Testing and Publication from 'dev'
+
+on: [push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  run-linter:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version: ["3.8.6"]
+
+    steps:
+      - name: Checkout this repository
+        uses: actions/checkout@v3
+        with:
+          lfs: true
+
+      - name: Checkout actions repository
+        uses: actions/checkout@v3
+        with:
+          repository: Exabyte-io/actions
+          token: ${{ secrets.BOT_GITHUB_TOKEN }}
+          path: actions
+
+      - name: Run ruff linter
+        uses: ./actions/py/lint
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run pre-commit hooks
+        run: |
+          python -m pip install pre-commit
+          pre-commit run --all
```

Comment:
Ruff is included in the pre-commit hooks. Perhaps we can change the "lint" action here to run all the pre-commit hooks instead. https://github.com/Exabyte-io/actions/blob/main/py/lint/action.yml

Reply:
Correct, I was thinking between extending the existing `./actions/py/lint` and adding the checks here. With pre-commit, we can specify all but the `ruff` check, but I'd rather do it consistently via the pre-commit.